lib/loading.js

```diff
@@ -44,25 +44,38 @@ class LoadingUI {
 
   get state () {
     return {
+      error: this.error,
       states: this.states,
       allDone: this.allDone,
       hasErrors: this.hasErrors
     }
   }
 
   setStates (states) {
+    this.clearError()
     this.states = states
     this.allDone = this.states.every(state => state.progress === 0 || state.progress === 100)
     this.hasErrors = this.states.some(state => state.hasErrors === true)
     this.broadcastState()
   }
 
-  setErrorState (error) {
-    this.setStates([{
-      name: 'error',
-      error: `${error}`,
-      stack: `${parseStack(error.stack).join('\n')}`
-    }])
+  setError (error) {
+    this.clearStates(true)
+    this.error = {
+      description: error.toString(),
+      stack: parseStack(error.stack).join('\n')
+    }
+    this.broadcastState()
+  }
+
+  clearError () {
+    this.error = undefined
+  }
+
+  clearStates (hasErrors) {
+    this.status = []
```

Comment:
@pi0 oops, made a typo here! should be states not status

Reply:
Would you please make direct commit to master?

Reply:
this message was to ping you asap in case you were already releasing ;)

Reply:
I won't release without proper testing ;)